crates/wasmtime/src/store.rs

```diff
@@ -943,7 +941,7 @@ impl<T> Store<T> {
     /// for an introduction to epoch-based interruption.
     pub fn epoch_deadline_callback(
         &mut self,
-        callback: impl FnMut(StoreContextMut<T>) -> Result<u64> + Send + Sync + 'static,
+        callback: impl FnMut(StoreContextMut<T>) -> DeadlineResult + Send + Sync + 'static,
```

Comment:
From an ergonomic perspective could this remain `Result<DeadlineResult>`? (maybe with a bit of renaming)

That'll allow `?` to be used for fallible operations and matches how traps are handled elsewhere in the embedding API where it's the `Err` of a `Result`

Reply:
To be clear, this would remove the `DeadlineResult::Trap` variant right? Since that would become the `Result::Err`?

Reply:
Yeah that's what I'm thinking